product_document_domain/models/product_product.py

```diff
@@ -0,0 +1,39 @@
+# Copyright 2024 Alberto Martínez <alberto.martinez@sygel.es>
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+
+from odoo import models
+
+
+class ProductProduct(models.Model):
+    _inherit = "product.product"
+
+    def action_open_documents(self):
+        res = super().action_open_documents()
+        res["context"].update(
+            {
+                "default_res_model": self._name,
+                "default_res_id": self.id,
+                "default_parent_res_id": self.product_tmpl_id.id,
+                "search_default_context_variant": False,
+                "search_default_context_template": False,
+                "search_default_context_variant_template": True,
+            }
+        )
+        return res
+
+    def _compute_product_document_count(self):
```

Comment:
Have you thought of calling super and then adding the number of records that meet the domain  ("res_model", "=", "product.template"), ("res_id", "=", template.id)?